docs/decoupling.md

```diff
@@ -3,24 +3,30 @@ description: Instructions on running WildBeast from source instead of from Docke
+This document outlines the procedure for decoupling WildBeast from Docker, allowing the entire system to be leveraged as opposed to a Docker sandbox.
 
 ## Preface
 
-If you have already set up your WildBeast instance in accordance with the available installation guides, feel free to jump to the next section. If you have not done this yet, however, please follow through with that procedure before embarking onto this guide. Most of the steps assume knowledge of what has been done previously, and as such it is highly recommended to follow proper procedure first.
+If you have already set up your WildBeast instance in accordance with the available installation guides, feel free to jump to the next section. If you have not done this yet, however, please follow through with that procedure before embarking onto this guide.
 
-Additionally, be sure to shutdown all existing Docker containers pertaining to WildBeast before proceeding.
+Additionally, consider whether you need to go through the decoupling process to begin with. The Docker container for WildBeast runs the code that is in the WildBeast directory, and when the container is restarted, the changes are mirrored. However, if you want to use something akin to live reloading (Such as Nodemon) or have other valid reasons, feel free to proceed.
 
 ## Reconfiguring containers and WildBeast
 
+Shutdown all existing Docker containers pertaining to WildBeast before starting.
+
 !!! warning "Port conflicts"
-    If you have other services running on ports **8529** or **2333**, select different ports for the sections below. You may encounter errors or conflicts otherwise. However, **do not modify the Docker port** in contradiction to the guide, or you will have trouble reaching the containers in the desired manner.
+    If you have other services running on ports **8529**, **80** or **2333**, select other available ports in the sections below and update your configuration accordingly. This is to avoid port conflicts. However, **do not modify the Docker port** in contradiction to this guide, or you will have trouble reaching the containers in the desired manner.
+
+!!! bug "Exceptions for Docker Toolbox"
+    If you're using Docker Toolbox, the hostname for a service will not be **localhost**. Due to how Docker Toolbox handles networking, the hostname will be an IPv4 address, such as **192.168.99.100**. In this case, this address must be used instead. The IP can be found by going to Kitematic and checking the **Hostname / Ports** section of any container.
 
 First, open the **.env** configuration file in the directory to which you installed the WildBeast source code, and tweak these values as follows:
 
 ```bash
-LAVA_NODES=[{"host":"localhost","port":2333,"region":"us","password":"password"}]
 ARANGO_URI=http://localhost:8529
+LAVA_NODES=[{"host":"localhost","port":80,"region":"us","password":"password"}]
```

Comment:
why not have the LAVA_REST_PORT prop be part of the LAVA_NODES json?

Reply:
Because the `LAVA_NODES` environment variable is passed directly to the `eris-lavalink` constructor. This method avoids unnecessary data pollution.

In addition, it's far easier to reference. Consider the following:
```js
const restPort = process.env.LAVA_REST_PORT
// vs
const restPort = process.env.LAVA_NODES ? process.env.LAVA_NODES[0].restPort : '2333'
```

Reply:
Addition to my previous, if we wanted to be fully safe with that method, we'd need to resort to such black magic as nested ternaries or other witchcraft:

```js
const restPort = process.env.LAVA_NODES
  ? process.env.LAVA_NODES[0]
    ? process.env.LAVA_NODES[0].restPort
    : '2333'
  : '2333'
```

Considering the execution context as well, that is simply not practical. The current method works far better.
https://github.com/TheSharks/WildBeast/blob/b85d9723ff51bfaafc4e8578dd6b659b60351f8f/src/encoders/lavalink.js#L53, doing

Reply:
the constructor doesn't care about extra fields, the function would then be
```
superagent.get(`http://${nodes[0].host}:${nodes[0].rest_port}/loadtracks?identifier=${search}`)
```
which is neater than what's currently there, rest port is not referenced anywhere else

Reply:
there is no witchcraft, the port has to be filled in.

Reply:
I forgot we can just reference the `nodes` object directly. I'll have a second gander at that.